lua/ui/game/announcement/AbstractAnnouncement.lua

```diff
@@ -0,0 +1,421 @@
+            self.AnimateThreadInstance:Destroy()
+        end
+
+        self.AnimateThreadInstance = self.Trash:Add(ForkThread(self.AnimateThread, self, control))
+    end,
+
+    --- The default animation sequence for an announcement. The announcement is destroyed at the end of the animation.
+    ---@param self UIAbstractAnnouncement
+    ---@param control Control
+    AnimateThread = function(self, control)
+        -- internal configuration of the animation
+        local expandDuration = 0.4
+        local contractDuration = 0.4
+        local contentDuration = 2.0
+        local contentShowDuration = 0.6
+        local contentHideDuration = 0.4
+
+        -- expand animation
+        self:ExpandBackground(control, expandDuration)
+        WaitSeconds(expandDuration)
+        if IsDestroyed(self) then
+            return
+        end
+
+        -- show content and hide it
+        self:AnimateContent(contentShowDuration, 1.0)
+        WaitSeconds(contentShowDuration + contentDuration)
+        if IsDestroyed(self) then
+            return
+        end
+
+        self:AnimateContent(contentHideDuration, 0.0)
+        WaitSeconds(contentHideDuration)
+        if IsDestroyed(self) then
+            return
+        end
+
+        -- contract animation
+        self:ContractBackground(control, contractDuration)
+        WaitSeconds(contractDuration)
+        if IsDestroyed(self) then
+            return
+        end
+
+        self:Destroy()
+    end,
+
+    --- Animates the alpha value of the content area.
+    ---@param self UIAbstractAnnouncement
+    ---@param duration number
+    ---@param targetTransparency number
+    AnimateContent = function(self, duration, targetTransparency)
+        -- do not allow other animations
+        if self.AnimateContentThreadInstance then
+            self.AnimateContentThreadInstance:Destroy()
+        end
+
+        local from = self.ContentArea:GetAlpha()
+        self.AnimateContentThreadInstance = self.Trash:Add(ForkThread(self.AnimateContentThread, self, duration, from, targetTransparency))
+    end,
+
+    --- Animates the alpha value of the content area.
+    ---@param self UIAbstractAnnouncement
+    ---@param duration number
+    ---@param target number
+    AnimateContentThread = function(self, duration, from, target)
+        -- animate it
+        local startTime = GetSystemTimeSeconds()
+        local endTime = startTime + duration
+        while not IsDestroyed(self) do
+            local currentTime = GetSystemTimeSeconds()
+            if currentTime > endTime then
+                break
+            end
+
+            local progress = (currentTime - startTime) / duration
+            local alpha = math.clamp(MATH_Lerp(progress, from, target), 0, 1)
+            self.ContentArea:SetAlpha(alpha, true)
+
+            WaitFrames(1)
+        end
+    end,
+
+    --- Expands the background of the announcement, starting at the provided control towards the center of the screen.
```

Comment:
Should be changed in other comments too.
```suggestion
    --- Expands the background of the announcement, starting at the provided control towards the content area.
```